### PR TITLE
C++: add missing 1.19 change notes

### PR DESCRIPTION
**Pull request by jbj**
With this PR, I think the change notes cover all the improvements we've made from 1.18 to 1.19. I've looked through all the PRs against this repo by myself, @dave-bartolomeo and @rdmarsh2 to see what was missing from the change notes.

The next step is that @felicity-semmle edits the change notes for consistency, language, etc.

### CODE REVIEW

**Comment by jbj** [comment]
I just sneaked a deprecation notice into this PR, along with a change note for it. There's unfortunately no way to deprecate a `qll` file, so the deprecation is in text only.

File: change-notes/1.19/analysis-cpp.md [hunk 1]
```diff
@@ -23,8 +23,12 @@
...
 | Resource not released in destructor | Fewer false positive results | Placement new is now excluded from the query. Also fixed an issue where false positives could occur if the destructor body was not in the snapshot. |
+| Memory is never freed | Fewer false positive results | This query now accounts for C++ _placement new_, which returns a pointer that does not need to be freed. |
 | Missing return statement (`cpp/missing-return`) | Visible by default | The precision of this query has been increased from 'medium' to 'high', which makes it visible by default in LGTM. It was 'medium' in release 1.17 and 1.18 because it had false positives due to an extractor bug that was fixed in 1.18. |
 | Missing return statement | Fewer false positive results | The query is now produces correct results when a function returns a template-dependent type, or makes a non-returning call to another function. |
+| Multiplication result converted to larger type (`cpp/integer-multiplication-cast-to-long`) | Fewer false positive results | Char-typed numbers are no longer considered to potentially large. |
```
Comment (geoffw0) [inline change-notes/1.19/analysis-cpp.md]: `to` -> `to be`?

File: change-notes/1.19/analysis-cpp.md [hunk 2]
```diff
@@ -23,8 +23,12 @@
...
+| Non-virtual destructor in base class (`cpp/virtual-destructor`) | Fewer false positive results | This query was renamed from "No virtual destructor" and moved from file name `AV Rule 78.ql` to `NonVirtualDestructorInBaseClass.ql`. The new version ignores base classes with non-public destructors since we consider those to be adequately protected. |
```
Comment (geoffw0) [inline change-notes/1.19/analysis-cpp.md]: I don't like the phrasing "renamed" and "moved" here as it suggests that `AV Rule 78.ql` no longer exists.  I think we should say that it was "copied".

Comment (jbj) [reply change-notes/1.19/analysis-cpp.md]: I agree. I also find my explanation here to be more convoluted that it ought to be.

Comment (jbj) [reply change-notes/1.19/analysis-cpp.md]: I've attempted to update the explanation now. I made it more precise but possibly even more convoluted.

**Comment by geoffw0** [comment]
:+1:

Presumably this needs a review from someone in @Semmle/doc.

**Pull request merged**